OpenEdXMobile/src/main/java/org/edx/mobile/player/PlayerFragment.java

```diff
@@ -1437,12 +1441,53 @@ public void onClick(View paramAnonymousView) {
                     tv_closedCaption.setBackgroundResource(R.drawable.grey_roundedbg);
                     tv_closedCaption.setOnClickListener(null);
                 }
+
+                TextView tvVideoSpeed = (TextView) layout.findViewById(R.id.tv_video_speed);
+                tvVideoSpeed.setOnClickListener(new OnClickListener() {
+                    @Override
+                    public void onClick(View v) {
+                        showVideoSpeedFragmentPopup();
+                    }
+                });
             }
         }catch(Exception e){
             logger.error(e);
         }
     }
 
+    /**
+     * This function is used to show Dialog fragment of
+     * Video playback speed list
+     */
+    private void showVideoSpeedFragmentPopup() {
+        hideSettingsPopUp();
+        List<Float> speeds = new ArrayList<>();
+        speeds.add(0.5f);
+        speeds.add(1.f);
+        speeds.add(1.5f);
+        speeds.add(2.0f);
```

Comment:
These shouldn't be here. These values belong inside the `SpeedDialogFragment` as constants.

Bonus point: It'd be really great if you write a separate FloatDescriptor interface for these constants for build time checking.

Reply:
@miankhalid are these okay as playback speed constants?
```
SLOW_PLAYBACK_SPEED = 0.5f;
NORMAL_PLAYBACK_SPEED = 1.0f;
FAST_PLAYBACK_SPEED = 1.5f;
VERY_FAST_PLAYBACK_SPEED = 2.0f;
```